docs/user/ppl/admin/datasources.rst

```diff
@@ -25,44 +25,91 @@ Definitions of datasource and connector
 
 Example Prometheus Datasource Definition ::
 
-    [{
+    {
         "name" : "my_prometheus",
         "connector": "prometheus",
         "properties" : {
             "prometheus.uri" : "http://localhost:8080",
             "prometheus.auth.type" : "basicauth",
             "prometheus.auth.username" : "admin",
             "prometheus.auth.password" : "admin"
-        }
-    }]
+        },
+        "allowedRoles" : ["prometheus_access"]
+    }
 Datasource configuration Restrictions.
 
 * ``name``, ``connector``, ``properties`` are required fields in the datasource configuration.
+* In case of secure domains, ``allowedRoles`` can be used to specify the opensearch roles allowed to access the datasource via PPL/SQL.
+* In case of secure domains, the plugin would only provide access to admin by default for PPL/SQL queries on datasources.
```

Comment:
what is the meaning of by default? without provide allowedRoles?

Reply:
yes. Should I rephrase this paragraph?